restful.module

```diff
@@ -573,11 +573,11 @@ function restful_date_time_format_element_validate($element, &$form_state) {
 function restful_restful_resource_alter(ResourceInterface &$resource) {
   $plugin_definition = $resource->getPluginDefinition();
   // Check for the rate limit configuration.
-  if (!empty($plugin_definition['rateLimit'])) {
-    $resource = new RateLimitedResource($resource);
-  }
-  // Check for the rate limit configuration.
   if (!empty($plugin_definition['renderCache']['render'])) {
     $resource = new CachedResource($resource);
   }
+  // Check for the rate limit configuration.
+  if (!empty($plugin_definition['rateLimit'])) {
```

Comment:
So rate limit needs to be the last one. IT means that implementing modules will need to alter the alter. We don't have to tackle this now, but worth thinking about a nicer DX